torch/cuda/__init__.py

```diff
@@ -487,7 +487,8 @@ def set_stream(stream: Stream):
         return
     torch._C._cuda_setStream(stream_id=stream.stream_id, device_index=stream.device_index, device_type=stream.device_type)
 
-def _parse_visible_devices() -> Set[int]:
+
+def _parse_visible_devices() -> Union[Set[int], List[str]]:
```

Comment:
Should always return a list (`Union[List[int], List[str]]`). The order matters.

Reply:
Sorry, not sure I understand why? Or why order matters

Reply:
Some other issue and PR want to query the GPU utilization by CUDA ordinals.

- #94472
- #94568

so we need to preserve the order. In #94568, the ordered ordinals are used. If we are only counting the number of devices, we can ignore the order here.